plugins/gradle/java/testSources/execution/test/runner/GradleRerunFailedTestsTest.kt

```diff
@@ -81,4 +81,54 @@ class GradleRerunFailedTestsTest : GradleRerunFailedTestsTestCase() {
       """.trimIndent())
     assertFalse(performRerunFailedTestsAction())
   }
-}
\ No newline at end of file
+
+  @Test
+  fun `test return failed tests abstract superclass`() {
+    createProjectSubFile("src/test/java/org/example/TestCase.java", """
+      package org.example;
+      import $jUnitTestAnnotationClass;
+      public abstract class TestCase {
+        @Test public void test1() {}
+        @Test public void test2() { throw new RuntimeException(); }
+        @Test public void test3() {}
+        @Test public void test4() { throw new RuntimeException(); }
+        @Test public void test5() { throw new RuntimeException(); }
+        @Test public void test6() {}
+        @Test public void test7() {}
+      }
+    """.trimIndent())
+    createProjectSubFile("src/test/java/org/example/SubTestCase.java", """
+      package org.example;
+      public class SubTestCase extends TestCase {
+      }
+    """.trimIndent())
+    createProjectSubFile("build.gradle", buildscript {
+      withJUnit()
+      withJavaPlugin()
+    })
+    importProject()
+
+    execute(":test --tests org.example.SubTestCase")
+    assertThat(getJUnitTestsExecutionTree())
+      .isEqualTo("""
+        -[root]
+         -SubTestCase
+          test1
+          test2
+          test3
+          test4
+          test5
+          test6
+          test7
+      """.trimIndent())
+    assertTrue(performRerunFailedTestsAction())
+    assertThat(getJUnitTestsExecutionTree())
+      .isEqualTo("""
+        -[root]
+         -SubTestCase
```

Comment:
Before the change, if `TestCase` is not abstract, this would say `TestCase` and list the tests. With it being abstract, it fails to execute anything and only lists `root`